localstack-core/localstack/services/lambda_/provider.py

```diff
@@ -326,9 +326,39 @@ def on_after_state_load(self):
                                 exc_info=True,
                             )
 
-                # Restore event source listeners
                 for esm in state.event_source_mappings.values():
-                    EventSourceListener.start_listeners_for_asf(esm, self.lambda_service)
+                    if config.LAMBDA_EVENT_SOURCE_MAPPING == "v2":
+                        # Restores event source workers
+                        function_arn = esm.get("FunctionArn")
+
+                        # TODO: How do we know the event source is up?
+                        # A basic poll to see if the mapped Lambda function is active/failed
+                        if not poll_condition(
+                            lambda: get_function_version_from_arn(function_arn).config.state.state
+                            in [State.Active, State.Failed],
+                            timeout=10,
+                        ):
+                            LOG.warning(
+                                "Creating ESM for Lambda that is not in running state: %s",
+                                function_arn,
+                            )
+
+                        function_version = get_function_version_from_arn(function_arn)
+                        function_role = function_version.config.role
+
+                        is_esm_enabled = esm.get("State", EsmState.DISABLED) not in (
+                            EsmState.DISABLED,
+                            EsmState.DISABLING,
```

Comment:
If the state is DISABLING when loading, we never set it to be disabled, even though the ESM worker is created correctly. That can lead to issues with the state later on, could we correct that?

Reply:
Ahh that is a good point. Let me just ensure we set the state to `DISABLED` if we encounter a `DISABLING` state on startup.

I took this approach since I assumed there was some async cleanups/actions that a worker would take prior to setting the state to `DISABLED`.

Reply:
The internal state of the EsmWorker is fine, but we never update the state in the store, right? So, the state in the store will be `DISABLING` indefinitely.

Reply:
Ahh I see. Will change this in the store

Reply:
Gave this another look. 

Appears we do check if the state is `DISABLING` in the `poller_loop`, where that `update_esm_state_in_store` updates the ESM state store `state.event_source_mappings`

https://github.com/localstack/localstack/blob/9b3530f8d2a996a3eebf6b4389812beef8f63129/localstack-core/localstack/services/lambda_/event_source_mapping/esm_worker.py#L149-L153

Reply:
I see that, but since we set `self.enabled` to `False`, and the .create method has this branch: https://github.com/localstack/localstack/blob/9b3530f8d2a996a3eebf6b4389812beef8f63129/localstack-core/localstack/services/lambda_/event_source_mapping/esm_worker.py#L84-L88

The poller loop never gets executed, and therefore the code you mentioned never called.

Reply:
A-ha! That makes sense. Will need to explicitly update this then

Reply:
I've added a call to the `update_esm_state_in_store` method in the `EsmWorker` class. This way, we actually set the state to `DISABLED` in the store when `create()` is called:

https://github.com/localstack/localstack/blob/ecf24ecab704417abadebe6ca158de6e1f18f52e/localstack-core/localstack/services/lambda_/event_source_mapping/esm_worker.py#L151-L154